### PR TITLE
[RFR] Graceful relative paths in CLI and config file

### PR DESCRIPTION
**Pull request by valeriangalliat**
This needs to be merged with `--no-ff` in both `master` and `develop` after review. Then `make publish` on master.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ Options:
   -h, --help            Bring help.
   --version             Show version.
   -v, --verbose         Enable verbose mode.
-  -d, --dest=<dir>      Documentation folder [default: sassdoc].
```
Comment (HugoGiraudel) [inline src/cli.js]: Why did you remove the default value here?

Comment (valeriangalliat) [inline src/cli.js]: See commit message and Slack discussion; it was overriding any config-defined `dest` even if not explicitely given via CLI.

File: src/environment.js [hunk 2]
```diff
@@ -135,6 +135,13 @@ export default class Environment extends EventEmitter {
       this.dir = process.cwd();
     }
 
+    if (!this.dest) {
+      this.dest = 'sassdoc';
+    }
+
+    this.dest = this.resolve(this.dest, this.destCli);
+    this.destName = path.relative(process.cwd(), this.dest);
```
Comment (pascalduez) [inline src/environment.js]: I find the keys `destName` and `themeName` a bit confusing, since they are actually paths. (relative but still).

File: src/environment.js [hunk 3]
```diff
@@ -289,9 +299,11 @@ export default class Environment extends EventEmitter {
    * Resolve given file from `this.dir`.
    *
    * @param {String} file
+   * @param {Boolean} cwd - whether it's relative to CWD (like when
+   *                        defined in CLI).
    * @return {String}
    */
-  resolve(file) {
-    return path.resolve(this.dir, file);
+  resolve(file, cwd = false) {
```
Comment (pascalduez) [inline src/environment.js]: Similarly here, I would rather have a "from CLI" or "CLI enabled" naming, rather than "themeCli" and "destCLI" which may lead to think there are 2 different things.

File: src/environment.js (hunk 2 above)
Comment (valeriangalliat) [inline src/environment.js]: You're right, though `themeName` can also be the package name (not path)… and it's used only for display.

We could call them `destPath` and `themePath` but it would not be relevant for `theme` in the case of a non-path package name.

I'm thinking the most correct way would be to rename them in `displayDest` and `displayTheme`.

But `themeName` was already present in the interface, I don't think it's documented though, so we can consider it private and might rename it without "breaking".

Comment (pascalduez) [inline src/environment.js]: Then yeah, either `displayDest` and `displayTheme`.
Or maybe a method `display()` which could be used like `env.display('themeName')` ?

Comment (valeriangalliat) [inline src/environment.js]: You mean `env.display('theme')`? Not possible since the display value for the theme is figured out at the same time as `theme`, depending if it's a package or a path. Or `env.display()` would just be a proxy to the `env.displayTheme` variable.

Comment (pascalduez) [inline src/environment.js]: Let's keep it the simpler and DRY possible :-)

**Comment by pascalduez** [comment]
[LGTM](http://www.lgtm.in/i/gsyL4g9fq)
